src/reanimated2/screenTransition/commonTypes.ts

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+import type {
+  MeasuredDimensions,
+  ShadowNodeWrapper,
+  SharedValue,
+} from '../commonTypes';
+
+export type PanGestureHandlerEventPayload = {
+  x: number;
+  y: number;
+  absoluteX: number;
+  absoluteY: number;
+  translationX: number;
+  translationY: number;
+  velocityX: number;
+  velocityY: number;
+};
+
+export type AnimatedScreenTransition = {
+  topScreenFrame: (
+    event: PanGestureHandlerEventPayload,
+    screenSize: MeasuredDimensions
+  ) => Record<string, unknown>;
+  belowTopScreenFrame: (
+    event: PanGestureHandlerEventPayload,
+    screenSize: MeasuredDimensions
+  ) => Record<string, unknown>;
+};
+
+export type GoBackGesture =
+  | 'swipeRight'
+  | 'swipeLeft'
+  | 'swipeUp'
+  | 'swipeDown'
+  | 'verticalSwipe'
+  | 'horizontalSwipe'
+  | 'twoDimensionalSwipe';
+
+export type ScreenTransitionConfig = {
+  stackTag: number;
+  belowTopScreenId: number | ShadowNodeWrapper;
+  topScreenId: number | ShadowNodeWrapper;
+  screenTransition: AnimatedScreenTransition;
+  sharedEvent: SharedValue<PanGestureHandlerEventPayload>;
+  startingGesturePosition: SharedValue<PanGestureHandlerEventPayload>;
+  onFinishAnimation?: () => void;
+  isTransitionCanceled: boolean;
+  goBackGesture: GoBackGesture;
+  screenDimensions: MeasuredDimensions;
+};
+
+export type RNScreensTurboModuleType = {
+  startTransition: (stackTag: number) => {
+    topScreenId: number | ShadowNodeWrapper;
+    belowTopScreenId: number | ShadowNodeWrapper;
+    canStartTransition: boolean;
+  };
+  updateTransition: (stackTag: number, progress: number) => void;
+  finishTransition: (stackTag: number, isCanceled: boolean) => void;
```

Comment:
Again, `isCanceled` or `isCancelled`?
```suggestion
  finishTransition: (stackTag: number, isCanceled: boolean) => void;
```